File: java/issue/src/solution/Backstage.java
package solution;

class Backstage extends Item {

    Backstage(String name, int sellIn, int quality) {
        super(name, sellIn, quality);
    }

    @Override
    protected void updateSellIn() {
        super.updateSellIn();
        if(sellIn <= 0){
            quality = 0;
        }
    }

    @Override
    protected void updateQuality() {
        if(sellIn <= 0){
            sellIn = 0;
        }else {
            if (sellIn <= 5) {
                if (quality < 50) {
                   quality = quality +3;
                }
            } else if (sellIn <= 10) {
                if (quality < 50) {
                    quality = quality +2;
                }
            }
        }
    }
}

File: javascript/src/solution/Backstage.js
const Item = require('./Item');

class Backstage extends Item {
    constructor(name, sellIn, quality) {
        super(name, sellIn, quality);
        let sellInKey = Object.getOwnPropertySymbols(this)[1];
        let qualityKey = Object.getOwnPropertySymbols(this)[2];

        let updateSellInKey = Object.getOwnPropertySymbols(this)[3];
        let updateQualityKey = Object.getOwnPropertySymbols(this)[4];

        this[updateSellInKey] = function updateSellIn() {
            if (this[sellInKey] <= 0) {
                this[qualityKey] = 0;
            }
        };

        this[updateQualityKey] = function updateQuality() {
            if (this[sellInKey] <= 0) {
                this[sellInKey] = 0;
            } else {
                if (this[sellInKey] <= 5) {
                    if (this[qualityKey] < 50) {
                        this[qualityKey] = this[qualityKey] + 3;
                    }
                } else if (this[sellInKey] <= 10) {
                    if (this[qualityKey] < 50) {
                        this[qualityKey] = this[qualityKey] + 2;
                    }
                }
            }
        };
    }
}

module.exports = Backstage;